[GTK] Remove support for OpenGL API in the web process

carlosgcampos opened the pull request:
#### cfe917fec45bf72c371087ece034feee8454f1b4
<pre>
[GTK] Remove support for OpenGL API in the web process
<a href="https://bugs.webkit.org/show_bug.cgi?id=257648">https://bugs.webkit.org/show_bug.cgi?id=257648</a>

Reviewed by Miguel Gomez.

We don&apos;t really need to keep supporting OpenGL API in the web process.
The GTK port is the only one still supporting it, so can we remove another
different code path compared to other ports. We use OpenGL ES
unconditionally in the web process, and whatever GTK uses by default in
the UI process.

* Source/ThirdParty/ANGLE/PlatformGTK.cmake: Remove the USE_OPENGL case.
* Source/ThirdParty/ANGLE/PlatformWPE.cmake: Ditto.
* Source/WebCore/CMakeLists.txt: Remove TemporaryOpenGLSetting.cpp from compilation.
* Source/WebCore/Headers.cmake: Remove deleted headers.
* Source/WebCore/PlatformWin.cmake: Remove TemporaryOpenGLSetting.cpp from compilation.
* Source/WebCore/html/canvas/WebGLVertexArrayObject.cpp:
(WebCore::WebGLVertexArrayObject::WebGLVertexArrayObject): Remove OPENGL_ES ifdef that was left after 230256@main.
* Source/WebCore/platform/cocoa/CoreVideoSoftLink.cpp: Ditto.
* Source/WebCore/platform/cocoa/CoreVideoSoftLink.h: Ditto.
* Source/WebCore/platform/graphics/OpenGLESShims.h: Removed.
* Source/WebCore/platform/graphics/OpenGLShims.cpp: Removed.
* Source/WebCore/platform/graphics/OpenGLShims.h: Removed.
* Source/WebCore/platform/graphics/egl/GLContext.cpp:
(WebCore::GLContext::getEGLConfig): Remove OPENGL_ES ifdefs.
(WebCore::GLContext::create): Ditto.
(WebCore::GLContext::createOffscreen): Ditto.
(WebCore::GLContext::createSharing): Ditto.
(WebCore::GLContext::createContextForEGLVersion): Ditto.
(WebCore::initializeOpenGLShimsIfNeeded): Deleted.
* Source/WebCore/platform/graphics/gstreamer/PlatformDisplayGStreamer.cpp:
(WebCore::PlatformDisplay::gstGLContext const): Remove OPENGL_ES ifdefs.
* Source/WebCore/platform/graphics/opengl/TemporaryOpenGLSetting.cpp: Removed.
* Source/WebCore/platform/graphics/opengl/TemporaryOpenGLSetting.h: Removed.
* Source/WebCore/platform/graphics/texmap/TextureMapperContextAttributes.cpp:
(WebCore::TextureMapperContextAttributes::get): Remove OPENGL_ES ifdefs.
* Source/WebCore/platform/graphics/texmap/TextureMapperGL.cpp:
(WebCore::TextureMapperGLData::~TextureMapperGLData): Remove OPENGL_ES ifdefs.
(WebCore::TextureMapperGLData::getVAO): Ditto.
(WebCore::TextureMapperGL::beginPainting): Ditto.
(WebCore::TextureMapperGL::endPainting): Ditto.
* Source/WebCore/platform/graphics/texmap/TextureMapperGLHeaders.h:
* Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp:
(WebCore::TextureMapperShaderProgram::create): Remove OPENGL_ES ifdefs.
* Source/WebKit/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU): Use the current context to fill EGL info for the UI process.
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp:
(WebKit::AcceleratedBackingStoreDMABuf::ensureGLContext): Remove OPENGL_ES ifdef.
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreWayland.cpp:
(WebKit::tryInitializeEGL): Use libepoxy to check required extensions.
(WebKit::AcceleratedBackingStoreWayland::ensureGLContext): Remove OPENGL_ES ifdef.
(WebKit::isEGLImageAvailable): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostTextureMapper.cpp:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsPlayStation.cmake:
* Source/cmake/OptionsWPE.cmake:
* Source/cmake/OptionsWin.cmake:

Canonical link: <a href="https://commits.webkit.org/265017@main">https://commits.webkit.org/265017@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/db1dd7aa36c1df20ccdf91f60cee6d212e6de33a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9182 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9463 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/9691 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/10838 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9104 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/9190 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11451 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9430 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11963 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/9329 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10277 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/7974 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/10994 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7573 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/8380 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/15843 "18 flakes 126 failures") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/7887 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/8675 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8527 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/11847 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/8799 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9027 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/7365 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/9363 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8256 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/2242 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12479 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/9597 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1086 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/8788 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2356 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

magomez commented on Source/WebCore/html/canvas/WebGLVertexArrayObject.cpp:
```diff
@@ -51,10 +51,6 @@ WebGLVertexArrayObject::~WebGLVertexArrayObject()
 WebGLVertexArrayObject::WebGLVertexArrayObject(WebGLRenderingContextBase& context, Type type)
     : WebGLVertexArrayObjectBase(context, type)
 {
-#if USE(OPENGL_ES)
```
Shouldn't we remove the remove the #if check here, but keep the code inside as we're using GLES?

carlosgcampos replied:
That's what I thought, but I think this is a leftover from the OPENGL_ES removal by apple. I don't think it's needed.

carlosgcampos committed:
[GTK] Remove support for OpenGL API in the web process
https://bugs.webkit.org/show_bug.cgi?id=257648

Reviewed by Miguel Gomez.

We don't really need to keep supporting OpenGL API in the web process.
The GTK port is the only one still supporting it, so can we remove another
different code path compared to other ports. We use OpenGL ES
unconditionally in the web process, and whatever GTK uses by default in
the UI process.

* Source/ThirdParty/ANGLE/PlatformGTK.cmake: Remove the USE_OPENGL case.
* Source/ThirdParty/ANGLE/PlatformWPE.cmake: Ditto.
* Source/WebCore/CMakeLists.txt: Remove TemporaryOpenGLSetting.cpp from compilation.
* Source/WebCore/Headers.cmake: Remove deleted headers.
* Source/WebCore/PlatformWin.cmake: Remove TemporaryOpenGLSetting.cpp from compilation.
* Source/WebCore/html/canvas/WebGLVertexArrayObject.cpp:
(WebCore::WebGLVertexArrayObject::WebGLVertexArrayObject): Remove OPENGL_ES ifdef that was left after 230256@main.
* Source/WebCore/platform/cocoa/CoreVideoSoftLink.cpp: Ditto.
* Source/WebCore/platform/cocoa/CoreVideoSoftLink.h: Ditto.
* Source/WebCore/platform/graphics/OpenGLESShims.h: Removed.
* Source/WebCore/platform/graphics/OpenGLShims.cpp: Removed.
* Source/WebCore/platform/graphics/OpenGLShims.h: Removed.
* Source/WebCore/platform/graphics/egl/GLContext.cpp:
(WebCore::GLContext::getEGLConfig): Remove OPENGL_ES ifdefs.
(WebCore::GLContext::create): Ditto.
(WebCore::GLContext::createOffscreen): Ditto.
(WebCore::GLContext::createSharing): Ditto.
(WebCore::GLContext::createContextForEGLVersion): Ditto.
(WebCore::initializeOpenGLShimsIfNeeded): Deleted.
* Source/WebCore/platform/graphics/gstreamer/PlatformDisplayGStreamer.cpp:
(WebCore::PlatformDisplay::gstGLContext const): Remove OPENGL_ES ifdefs.
* Source/WebCore/platform/graphics/opengl/TemporaryOpenGLSetting.cpp: Removed.
* Source/WebCore/platform/graphics/opengl/TemporaryOpenGLSetting.h: Removed.
* Source/WebCore/platform/graphics/texmap/TextureMapperContextAttributes.cpp:
(WebCore::TextureMapperContextAttributes::get): Remove OPENGL_ES ifdefs.
* Source/WebCore/platform/graphics/texmap/TextureMapperGL.cpp:
(WebCore::TextureMapperGLData::~TextureMapperGLData): Remove OPENGL_ES ifdefs.
(WebCore::TextureMapperGLData::getVAO): Ditto.
(WebCore::TextureMapperGL::beginPainting): Ditto.
(WebCore::TextureMapperGL::endPainting): Ditto.
* Source/WebCore/platform/graphics/texmap/TextureMapperGLHeaders.h:
* Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp:
(WebCore::TextureMapperShaderProgram::create): Remove OPENGL_ES ifdefs.
* Source/WebKit/Shared/CoordinatedGraphics/threadedcompositor/ThreadedCompositor.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
(WebKit::WebKitProtocolHandler::handleGPU): Use the current context to fill EGL info for the UI process.
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp:
(WebKit::AcceleratedBackingStoreDMABuf::ensureGLContext): Remove OPENGL_ES ifdef.
* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreWayland.cpp:
(WebKit::tryInitializeEGL): Use libepoxy to check required extensions.
(WebKit::AcceleratedBackingStoreWayland::ensureGLContext): Remove OPENGL_ES ifdef.
(WebKit::isEGLImageAvailable): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostTextureMapper.cpp:
* Source/cmake/OptionsGTK.cmake:
* Source/cmake/OptionsPlayStation.cmake:
* Source/cmake/OptionsWPE.cmake:
* Source/cmake/OptionsWin.cmake:

Canonical link: https://commits.webkit.org/265017@main
